tests/test_api_v1.py

```diff
@@ -1488,7 +1488,90 @@ def test_fetch_request_content_manifest(
         assert rv.json == {"error": err_msg}
 
 
-@pytest.mark.parametrize("pkg_type", ["npm", "unknown", "gomod"])
+@pytest.mark.parametrize("state", ["complete", "stale", "in_progress", "failed"])
+def test_fetch_request_content_manifest_npm(app, client, db, auth_env, worker_auth_env, state):
+    data = {
+        "repo": "https://github.com/release-engineering/console-ui.git",
+        "ref": "c50b93a32df1c9d700e3e80996845bc2e13be848",
+        "pkg_managers": ["npm"],
+    }
+    # flask_login.current_user is used in Request.from_json, which requires a request context
+    with app.test_request_context(environ_base=auth_env):
+        request = Request.from_json(data)
+    request.add_state(state, "Some state")
+    db.session.add(request)
+    db.session.commit()
+
+    npm_pkgs = [
+        {"name": "client", "type": "npm", "version": "1.0.0"},
+        {"name": "proxy", "type": "npm", "version": "1.0.0"},
+    ]
+    npm_deps = [
+        {"dev": True, "name": "rxjs", "replaces": None, "type": "npm", "version": "6.5.5"},
+        {"dev": True, "name": "safe-regex", "replaces": None, "type": "npm", "version": "1.1.0"},
+        {"dev": True, "name": "rxjs", "replaces": None, "type": "npm", "version": "6.5.5"},
+        {"dev": False, "name": "react", "replaces": None, "type": "npm", "version": "16.13.1"},
+    ]
+    payload = {
+        "dependencies": npm_deps[:2],
+        "package": npm_pkgs[0],
+    }
+    client.patch(f"/api/v1/requests/1", json=payload, environ_base=worker_auth_env)
+    payload = {
+        "dependencies": npm_deps[2:],
+        "package": npm_pkgs[1],
+    }
+    client.patch(f"/api/v1/requests/1", json=payload, environ_base=worker_auth_env)
+
+    rv = client.get("/api/v1/requests/1")
+    assert rv.status_code == 200
+
+    # set expectations
+    json_schema_url = (
+        "https://raw.githubusercontent.com/containerbuildsystem/atomic-reactor/"
+        "f4abcfdaf8247a6b074f94fa84f3846f82d781c6/atomic_reactor/schemas/content_manifest.json"
+    )
+    npm_pkg_purls = [Package.from_json(pkg).to_purl() for pkg in npm_pkgs]
+
+    image_contents = [
+        {"purl": npm_pkg_purls[0], "dependencies": [], "sources": []},
+        {"purl": npm_pkg_purls[1], "dependencies": [], "sources": []},
+    ]
+
+    for i, d in enumerate(npm_deps):
+        pkg_index = 0
+        if i >= 2:
+            pkg_index = 1
+        d.pop("replaces")
+        is_dev = d.pop("dev")
+        p = Package.from_json(d).to_purl()
+        image_contents[pkg_index]["sources"].append({"purl": p})
+        if not is_dev:
+            image_contents[pkg_index]["dependencies"].append({"purl": p})
+
+    expected = {
+        "metadata": {"icm_version": 1, "icm_spec": json_schema_url, "image_layer_index": -1},
+        "image_contents": image_contents,
```

Comment:
Instead of dynamically creating the `image_contents` value that is expected, could you just directly compare against a static dictionary? I know it's more lines, but it ensures that the functions such as `to_purl` that are used to generate this value are returning the correct value.

It also makes it easier to read.

Reply:
Done